### DISCUSSION TITLE
Read d ts schema

### DISCUSSION THREAD

**Discussion by loremipsumowski** [microsoft/monaco-editor#3844] [Q&A]
Is there any way to read schema of d.ts file with Monaco by custom purposes? I would like to create custom completion items provider that is based on d.ts schema, but I would like to extend that items by adding detailed descriptions from external source.
Is that possible to modify suggestions made by Monaco, or simply read model of custom class added as external lib?

For example I have code:
```
declare class MyClass {
    /**
     * that is property
     */
    property: number;
}

const myClass = new MyClass();
myClass.
```

now, after ctrl+space I see there is `property` of number type, and any description, but I would like to add more description there myself, could I do it any way? Or could I extract list of properties and methods of class as JSON to use it myself?
![image](https://user-images.githubusercontent.com/10393760/225826192-c42f4b25-fdc8-4aa7-aa4f-71b95f0eb6a6.png)

**Comment by tamayika** [comment]
I don't know what description you want to add more, but you can write markdown.

```ts
declare class MyClass {
    /**
     * I can write markdown
     * # header
     * * item1
     * * item2
     * |col|col|
     * |---|---|
     * |d1 | d2|
     */
    property: number;
}
```
